test cases/rust/17 staticlib link staticlib/meson.build

```diff
@@ -0,0 +1,6 @@
+project('staticlib link staticlib', 'c', 'rust')
+
+leaf = static_library('leaf', 'leaf.rs', rust_crate_type : 'rlib')
+branch = static_library('branch', 'branch.rs', link_with: leaf, rust_crate_type : 'staticlib', install : true)
+e = executable('prog', 'prog.c', link_with : branch, install : true)
```

Comment:
The description talks of link_whole but neither of these two is defined as link_whole. Is that intentional? If yes, there should be comment explaining what is happening here.

Reply:
It was intentional. I've added a comment explaining what is going on there. (For those who might be reading this issue PR later, the link_with gets promoted to link_whole, which wasn't handled properly in the ninja backend for Rust targets).